### PR TITLE
[dua] nits and fix unexpected re-registration in #6538

### LINKED ISSUE #6538: [github-actions] test_ndproxy.py fails intermittently
[logs_72922.zip](https://github.com/openthread/openthread/files/6402091/logs_72922.zip)
[thread-1-2-backbone-results.zip](https://github.com/openthread/openthread/files/6402092/thread-1-2-backbone-results.zip)

### PR DESCRIPTION
**Pull request by librasungirl**
The last commit is to fix the unexpected re-registration behavior in #6538 
There are some other minor nits during issue debugging, submitted together this this fix. May keep separate commits when merge.

### CODE REVIEW

File: src/core/thread/dua_manager.cpp [hunk 1]
```diff
@@ -307,6 +307,8 @@ void DuaManager::HandleNotifierEvents(Events aEvents)
 {
     Mle::MleRouter &mle = Get<Mle::MleRouter>();
 
+    VerifyOrExit(mle.IsAttached());
```
Comment (simonlingoogle) [inline src/core/thread/dua_manager.cpp:310]: Should we stop the reregistration/registration timer when the node is detached?

Comment (librasungirl) [reply src/core/thread/dua_manager.cpp]: In `void DuaManager::HandleTimeTick(void)`, the registration is updated per ticker
https://github.com/openthread/openthread/blob/8dc9a53cc7530495023e10949cbafb4c4d18ae76/src/core/thread/dua_manager.cpp#L396
Updated to `VerifyOrExit(mle.IsAttached(), mDelay.mValue = 0);` to reset the delay value when detached, and ensure it is stopped.
Pls take another review.

File: src/core/thread/dua_manager.cpp [hunk 2]
```diff
@@ -307,6 +307,8 @@ void DuaManager::HandleNotifierEvents(Events aEvents)
 {
     Mle::MleRouter &mle = Get<Mle::MleRouter>();
 
+    VerifyOrExit(mle.IsAttached(), mDelay.mValue = 0);
```
Comment (simonlingoogle) [inline src/core/thread/dua_manager.cpp:310]: nit: should we call UpdateTimeTickerRegistration() immediately?

File: src/core/thread/dua_manager.cpp [hunk 3]
```diff
@@ -519,6 +524,7 @@ void DuaManager::PerformNextRegistration(void)
 
     mIsDuaPending   = true;
     mRegisteringDua = dua;
+    mDelay.mValue   = 0;
```
Comment (simonlingoogle) [inline src/core/thread/dua_manager.cpp:527]: Should we call UpdateTimeTickerRegistration() immediately?

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 

with nits

File: src/core/thread/dua_manager.cpp (hunk 2 above)
Comment (librasungirl) [reply src/core/thread/dua_manager.cpp:310]: It should be acceptable to have one extra `HandleTimeTick()` run and unregister time ticker receiver there? @simonlingoogle

File: src/core/thread/dua_manager.cpp (hunk 3 above)
Comment (librasungirl) [reply src/core/thread/dua_manager.cpp:527]: ditto

File: src/core/thread/dua_manager.cpp (hunk 2 above)
Comment (simonlingoogle) [reply src/core/thread/dua_manager.cpp:310]: Should be fine.

**Pull request merged**